### ISSUE TITLE
pg_active_session_history.query of dml query is not correct when blocked by other session executing ddl query like truncate or drop 

### ISSUE THREAD

**Issue by einisadog** [pgsentinel/pgsentinel#34]
I found some issue during my use of this extension that is when update query is blocked by truncate query, the pg_active_session_history.query column of the update query is not correct, and the correct text appears in  pg_active_session_history.top_level_query column. I think it's not quite right with this.

Thanks,
Ein

**Here is the my test result:**

postgres=# select datname,pid,wait_event_type,wait_event,state,backend_xid,top_level_query,query,queryid,blockers,blockerpid from pg_active_session_history                                                                                                 where ash_time = (select case when coalesce(extract(epoch from(now()-max(ash_time)))::float,0) < (select setting::float*2 from pg_settings where name='pgsentinel_ash.sampling_period') then max(ash_time) else null end 
from pg_active_session_history);
-[ RECORD 1 ]---+----------------------------------
datname         | db1
pid             | 9836
wait_event_type | Client
wait_event      | ClientRead
state           | idle in transaction
backend_xid     | 589
top_level_query | truncate table t1;
query           | truncate table t1
queryid         | -3182215069190252785
blockers        | 
blockerpid      | 
-[ RECORD 2 ]---+----------------------------------
datname         | db1
pid             | 9838
wait_event_type | Lock
wait_event      | relation
state           | active
backend_xid     | 
top_level_query | update t1 set v1=1111 where id=1;
query           | begin
queryid         | 3422818749220588372
blockers        | 1
blockerpid      | 9836

postgres=# select query from pg_stat_statements where queryid=3422818749220588372;                                                      -[ RECORD 1 ]
query | begin

**Comment by bdrouvot** [comment]
Indeed, that's because for a blocked session the `post_parse_analyze_hook` hook (that pgsentinel uses to populate the `query` field) has not been triggered yet.

**Comment by einisadog** [comment]
Thanks for your response. 
I found this issue only when dml query blocked by ddl query. When dml query blocked by other dml query or ddl query blocked by dml query, there is no problem.
I wonder if you have plan to fix this issue? 
Thanks very much.

postgres=# select datname,pid,wait_event_type,wait_event,state,backend_xid,top_level_query,query,queryid,blockers,blockerpid from pg_active_session_history where ash_time = (select case when coalesce(extract(epoch from(now()-max(ash_time)))::float,0) < (select setting::float*2 from pg_settings where name='pgsentinel_ash.sampling_period') then max(ash_time) else null end
postgres(# from pg_active_session_history);
-[ RECORD 1 ]---+----------------------------------
datname         | db1
pid             | 10206
wait_event_type | Client
wait_event      | ClientRead
state           | idle in transaction
backend_xid     | 498
top_level_query | update t1 set v1=111 where id=1;
query           | update t1 set v1=111 where id=1
queryid         | 68542832220132722
blockers        | 
blockerpid      | 
-[ RECORD 2 ]---+----------------------------------
datname         | db1
pid             | 10209
wait_event_type | Lock
wait_event      | transactionid
state           | active
backend_xid     | 499
top_level_query | update t1 set v1=1111 where id=1;
query           | update t1 set v1=1111 where id=1
queryid         | 68542832220132722
blockers        | 1
blockerpid      | 10206
